src/view/components/common/login-as/login-as.ts

```diff
@@ -17,20 +19,29 @@ import {Router} from '@ngrx/router';
 export class LoginAsComponent {
     @Output() cancel = new EventEmitter<>();
 
-    private userLists: Array<User>;
+    private i18nKeys: Array<string> = ['change', 'cancel', 'password'];
+    private i18nMessages = {};
+    private messageMapSubscription;
     private needPassword: boolean = false;
+    private userLists: Array<User>;
 
-    constructor(private loginService: LoginService, private router: Router,
-                @Inject('dotcmsConfig') private dotcmsConfig: DotcmsConfig) {
+    constructor(private loginService: LoginService, private router: Router, private messageService: MessageService) {
+        this.messageMapSubscription = this.messageService.getMessages(this.i18nKeys).subscribe(res => {
+            this.i18nMessages = _.pick(res, this.i18nKeys);
+        });
     }
 
     ngOnInit(): void {
-        this.loginService.loadLoginAsUsers();
-        this.loginService.loginAsUsers$.subscribe(data => {
+        this.loginService.loadLoginAsUsersList();
+        this.loginService.loginAsUsersList$.subscribe(data => {
```

Comment:
Maybe we could have just one method, the loginAsUsersList$ could check if the list is already loaded and if not then load it.